src/ashell/file-utils.c

```diff
@@ -170,3 +170,38 @@ int fs_get_boot_cfg_filename(const char *timestamp, char *filename)
     fs_close_alloc(file);
     return -1;
 }
+
+char *read_file_alloc(const char *file_name, ssize_t *size)
+{
+    char *file_buf = NULL;
+    fs_file_t *fp = fs_open_alloc(file_name, "r");
+
+    if (fp == NULL)
+        return 0;
+
+    *size = fs_size(fp);
+    if (*size == 0) {
+        ERR_PRINT("[ERR] Empty file (%s)\n", file_name);
+        fs_close_alloc(fp);
+        return NULL;
+    }
+
+    file_buf = (char *)zjs_malloc(*size);
+    if (file_buf == NULL) {
+        ERR_PRINT("[ERR] Not enough memory for (%s)\n", file_name);
+        fs_close_alloc(fp);
+        return NULL;
+    }
+
+    ssize_t brw = fs_read(fp, file_buf, *size);
+
+    if (brw != *size) {
+        ERR_PRINT("[ERR] Not enough memory for (%s)\n", file_name);
```

Comment:
This condition doesn't mean out of memory, it means the read didn't find enough bytes, or there was a read error (negative return value). The former shouldn't ever occur since you just got the file size. Still not bad to check but don't say out of memory. Maybe "read failed". Should be very uncommon so might make sense to put in DBG_PRINT only, but otoh kind of important if it did happen, so I'm of two minds about it.

Reply:
Created a macro here for the error and simplified it to just "Read file failed".  I've added DBG_PRINT as well in case someone needs to know more, it can be handy to know which file is failing and why.